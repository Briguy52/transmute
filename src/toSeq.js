// @flow
import { Seq } from 'immutable';
import type { Iterable } from 'immutable';

/**
 * Converts `subject` to a `Seq` if possible.
 *
 * @param  {Array|Iterable|Object|String} subject
 * @return {Seq}
 */
export default function toSeq<K, V>(
  subject: Array<V> | Iterable<K, V> | { [key: K]: V } | String
) {
  return Seq(subject);
}
