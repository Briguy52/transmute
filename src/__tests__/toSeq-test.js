import toSeq from "../toSeq";

describe("transmute/toSeq", () => {
  it("converts to a sequence", () => {
    expect(toSeq([1, 2, 3])).toMatchSnapshot();
    expect(toSeq("123")).toMatchSnapshot();
  });
});
