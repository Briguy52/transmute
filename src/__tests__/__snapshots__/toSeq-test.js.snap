// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/toSeq converts to a sequence 1`] = `
Immutable.Seq [
  1,
  2,
  3,
]
`;

exports[`transmute/toSeq converts to a sequence 2`] = `
Immutable.Seq [
  "1",
  "2",
  "3",
]
`;
